### PR TITLE
Fix unit test build issues and test failures for IP 

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR fixes unit test build issues and test failures for IP timers unit tests which were failing after the latest IPv6 integration changes. Creates 2 new diffconfig for IP - diffconfig1 and diffconfig2 for enabling/disabling `ipconfigCOMPATIBLE_WITH_SINGLE` for the currently existing IP diffconfigs.

Also fixes:
- Build issue with IP utils diffconfig
- Minor fix with an ARP unit test - `test_vARPRefreshCacheEntry_IPAndMACInDifferentLocations1`

Test Steps
-----------
```
cmake -S test/unit-test -B test/unit-test/build/ 
make -C test/unit-test/build/ all 
cd test/unit-test/build/
ctest -E system --output-on-failure --verbose

```
Related Issue
-----------
IP unit test cases were failing due to the changes in the source code as part of the IPv6 integration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/FreeRTOS_IP_DiffConfig/FreeRTOS_IP_DiffConfig_utest.c [hunk 1]
```diff
@@ -115,38 +115,38 @@ static size_t StubuxStreamBufferGetPtr_ReturnCorrectVals( StreamBuffer_t * pxBuf
...
 }
 
-static void vSetIPTaskHandle( TaskHandle_t xTaskHandleToSet )
-{
-    const uint8_t ucIPAddress[ ipIP_ADDRESS_LENGTH_BYTES ];
-    const uint8_t ucNetMask[ ipIP_ADDRESS_LENGTH_BYTES ];
-    const uint8_t ucGatewayAddress[ ipIP_ADDRESS_LENGTH_BYTES ];
-    const uint8_t ucDNSServerAddress[ ipIP_ADDRESS_LENGTH_BYTES ];
-    const uint8_t ucMACAddress[ ipMAC_ADDRESS_LENGTH_BYTES ];
+// static void vSetIPTaskHandle( TaskHandle_t xTaskHandleToSet )
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_IP_DiffConfig/FreeRTOS_IP_DiffConfig_utest.c:118]: Can we remove if this test is not needed?

Comment (tony-josi-aws) [reply test/unit-test/FreeRTOS_IP_DiffConfig/FreeRTOS_IP_DiffConfig_utest.c:118]: sure

File: test/unit-test/FreeRTOS_IP_DiffConfig/FreeRTOS_IP_DiffConfig_utest.c [hunk 2]
```diff
@@ -257,57 +257,57 @@ static BaseType_t StubxTaskCreate( TaskFunction_t pxTaskCode,
...
-void test_FreeRTOS_IPInit_HappyPath( void )
-{
-    const uint8_t ucIPAddress[ ipIP_ADDRESS_LENGTH_BYTES ] = { 0xC0, 0xB0, 0xAB, 0x12 };
-    const uint8_t ucNetMask[ ipIP_ADDRESS_LENGTH_BYTES ] = { 0xC1, 0xB2, 0xAC, 0x13 };
-    const uint8_t ucGatewayAddress[ ipIP_ADDRESS_LENGTH_BYTES ] = { 0xC2, 0xB3, 0xAC, 0x14 };
-    const uint8_t ucDNSServerAddress[ ipIP_ADDRESS_LENGTH_BYTES ] = { 0xC3, 0xB4, 0xAD, 0x15 };
-    const uint8_t ucMACAddress[ ipMAC_ADDRESS_LENGTH_BYTES ] = { 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF };
-    BaseType_t xReturn;
-    QueueHandle_t ulPointerToQueue = ( QueueHandle_t ) 0x1234ABCD;
+// void test_FreeRTOS_IPInit_HappyPath( void )
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_IP_DiffConfig/FreeRTOS_IP_DiffConfig_utest.c:260]: why is this one removed? this means everything working correctly right? Does we not need similar test case?

Comment (tony-josi-aws) [reply test/unit-test/FreeRTOS_IP_DiffConfig/FreeRTOS_IP_DiffConfig_utest.c]: this particular tests IPConfig file disables `ipconfigCOMPATIBLE_WITH_SINGLE` hence `FreeRTOS_IPInit` wont be available, so the test is moved to other diffconfig1 and diffconfig2

**Review by moninom1 — APPROVED** [review]

**Pull request merged**